src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -2222,43 +2222,45 @@ antlrcpp::Any CypherMainVisitor::visitNumberLiteral(MemgraphCypher::NumberLitera
 }
 
 antlrcpp::Any CypherMainVisitor::visitFunctionInvocation(MemgraphCypher::FunctionInvocationContext *ctx) {
-  if (ctx->DISTINCT()) {
-    throw utils::NotYetImplemented("DISTINCT function call");
-  }
   auto function_name = std::any_cast<std::string>(ctx->functionName()->accept(this));
   std::vector<Expression *> expressions;
   for (auto *expression : ctx->expression()) {
     expressions.push_back(std::any_cast<Expression *>(expression->accept(this)));
   }
   if (expressions.size() == 1U) {
     if (function_name == Aggregation::kCount) {
-      return static_cast<Expression *>(storage_->Create<Aggregation>(expressions[0], nullptr, Aggregation::Op::COUNT));
+      return static_cast<Expression *>(
+          storage_->Create<Aggregation>(expressions[0], nullptr, Aggregation::Op::COUNT, ctx->DISTINCT()));
```

Comment:
It is just a small detail, but maybe try to save this value instead of querying in in every case. `const auto is_distinct = ctx->DISTINCT() != nullptr;` should work in my opinion.